### ISSUE TITLE
Makeflow: Don't redirect to stdout

### ISSUE THREAD

**Issue by dthain** [cooperative-computing-lab/cctools#2701]
Matt Hayman reports:

"NCAR has recently imposed a new rule that jobs submitted to the Cheyenne cluster cannot direct output to /dev/null.  The job is killed as soon as the event is detected.  In investigating our software, my colleague and I have determined that makeflow appears to be redirecting some output to /dev/null.  Is there some way that we can redirect the output that makeflow is sending to /dev/null to a temporary file?  Is there any known work around for this issue?"

So, we need to fix up batch_job so that the wrapper doesn't send output to /dev/null, but instead to another file.  (And then we need to clean it up somehow.)

**Comment by dthain** [comment]
The /dev/null comes from the wrapper script.  It's not really a problem b/c the user that wants to capture stdout of a makeflow task does so by redirecting the output in the command line itself.

We can hack this into place, but then the problem is that each job ends up creating a new output file that has to be cleaned up.   Thoughts?

https://github.com/cooperative-computing-lab/cctools/blob/40e3acbc21a5b62172367dd8e993285c8349bed8/batch_job/src/batch_job_cluster.c#L451

**Comment by btovar** [comment]
We can add a:

batch_queue_option_is_yes(q, "verbose-batch-commands");   // makeflow --verbose-batch-commands

so that any -o and -e specification is not redirected to /dev/null

**Comment by btovar** [comment]
The drawback of not redirecting it to /dev/null is that each job will get a job_name.ojob_id and a job_name.ejob_id.  We could also always  write to  some `snprintf(stdouterr, PATH_MAX, "%s.stdouterr", sysname);` file, as we do for the wrapper, and clean the file when closing the queues, etc.

**Comment by dthain** [comment]
Ok, I'm drafting a bit of code right now, then will pass it off to you...

**Comment by dthain** [comment]
See #2702 for a first crack at it -- please go ahead and take it over the finish line.

**Comment by dthain** [comment]
Is there something more to do on this one?

**Comment by btovar** [comment]
No, Matt has confirmed --keep-wrapper-stdout works for them, and the fix is included in cctools version 7.3.3.
